crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -77,6 +77,12 @@ impl<'a> SnapshotProviderRW<'a> {
         // TODO(joshie): update the index without re-iterating all snapshots
         self.reader.update_index()?;
 
+        // Reader needs to reload the static file again to have the new changes.
+        self.reader.remove_cached_provider(
+            self.writer.user_header().segment(),
+            *find_fixed_range(BLOCKS_PER_SNAPSHOT, self.writer.user_header().block_start()).end(),
+        );
+
```

Comment:
I think we should rather do `remove_cached_provider` in `reader.update_index`, so there's no chance to misuse this API. We'll need to pass the changed range in `update_index` then.